[green] Added method to get lat/long of a station

zackshapiro opened the pull request:
- Added `lat_and_long` method
- Associated tests for new method
- Fixed one `empty_stations` test to work with the sample data, moved one test to pending, unsure how to test it
- Bumped gemspec version to 0.0.65

zackshapiro committed:
updated tests for empty station, added an empty station to the request.rb file

zackshapiro commented:
@gabceb Wanna take a look at this one, +1 it if it looks good

gabceb commented on spec/bikeshare_spec.rb:
```diff
@@ -160,15 +160,32 @@
+  end
+
+  describe "#lat_and_long" do
+    it "should raise an error with a bad station id" do
```
This test case is already covered by https://github.com/zackshapiro/bikeshare/blob/master/spec/bikeshare_spec.rb#L28

gabceb commented on spec/bikeshare_spec.rb:
```diff
@@ -160,15 +160,32 @@
+
+    it "should raise an error with no station id" do
+      expect { @response.percent_available }.to raise_error
+    end
```
I think this and the previous test case should not be here. Maybe some copy && paste gone bad? :smirk:

gabceb commented:
I did not find the code associated with your commit message: "Fixed one empty_stations test to work with the sample data, moved one test to pending, unsure how to test it"

zackshapiro committed:
updated empty station (41) and offline station (38) test json

zackshapiro committed:
removed redundant tests

zackshapiro commented:
@gabceb Fixed those things you mentioned, thanks. Sample data has two stations, 38 which is offline and 41 which has zero bikes (it's empty). Made those changes in the test JSON file.

The pending test is at line 63 of `bikeshare_spec.rb`

gabceb commented:
This is what I see on line 63 when I pull your branch:

```
@response.stations("San Francisco").count.should == 34
```

gabceb commented:
One last thing, is usually good practice to bump versions on a separate commit so is easier for others to look if necessary. Versions bumps are also increased numerically according to the [SemVer spec](http://semver.org/)

zackshapiro commented:
Noted. Thanks. Not sure why you're not seeing appropriate changes. Try pulling again?

gabceb commented:
Have you pushed to this branch on Github? If you go to the "Files Changed" section of this PR I should be able to see your changes but still cannot see them.

zackshapiro commented:
Yeah, I have. I see the changes in my Files Changed section. Not sure why the discrepancy

gabceb commented:
Maybe you meant line 163? -> https://github.com/zackshapiro/bikeshare/pull/8/files#diff-ce1786ba1c822bdf89a438a3ec0a2528R163

zackshapiro commented:
Yep, my mistake

gabceb commented:
Ha. I was looking for an actual pending test like the one below. That way it will show on the tests results. Let me take a look at it see if I can help.

``` ruby
describe "an example" do
  it "is implemented but waiting" do
    pending("something else getting finished")
    this_should_not_get_executed
  end
end
```

zackshapiro commented:
Hah whoops, my bad. Thanks man

On Thu, Oct 10, 2013 at 2:47 PM, Gabriel Cebrian
notifications@github.comwrote:

> Ha. I was looking for an actual pending test like the one below. That way
> it will show on the tests results. Let me take a look at it see if I can
> help.
> 
> describe "an example" do
>   it "is implemented but waiting" do
>     pending("something else getting finished")
>     this_should_not_get_executed
>   endend
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/zackshapiro/bikeshare/pull/8#issuecomment-26094636
> .

gabceb committed:
Initialize method now accepts a url if passed
Added JSON response with offline stations
Move JSON responses to separate files
Added more detailed checks on the response of offline stations tests
Added test to check for url passed on initialize

gabceb commented:
The solution I propose is to add a url parameter to the initialize method to get this test to work. If you think this is a good idea you can pull my lat-long branch.

Here is my commit with the changes https://github.com/gabceb/bikeshare-1/commit/3b9518c6c3722afa0c2b0afd155cbb199b5f3fbf

This commit also make the json responses into separate files to keep the code clean.

zackshapiro commented:
@gabceb For the state of this gem right now and the needs that it feels, that seems excessive to alter the init function for the purposes of running tests alone. I'd rather merge this branch, put these changes on a shelf and revisit it in the future.

gabceb commented:
Allow me to disagree with you on this one @zackshapiro.

If you look at the commit code you will notice that the url passed is only used if available, otherwise the default url will be used so this won't break previous versions of the gem. If you think about it, the url is not added **only** for the test but as an option for developers to change that url if needed. The url parameter also happens to be used by one of the tests (actually 2 tests because I added a test for this change as well) :smirk:

Anyways, I do agree that this gem as it is does what it is supposed to do so don't feel bad if you don't want to pull my changes but I do think my changes are an improvement to the gem.

zackshapiro commented:
@gabceb After sleeping on it, I really like this. Was your branch branched off of my branch? I'd like to merge yours into master

gabceb commented:
Yeah. Just pull my branch lat-long from git@github.com:gabceb/bikeshare-1.git

zackshapiro commented:
Merged. Good work sir
